src/_blog/rust-libp2p-browser-to-server.md

```diff
@@ -0,0 +1,87 @@
+---
+tags:
+  - rust-libp2p
+  - rust
+  - libp2p
+  - browser
+  - webrtc
+  - webassembly
+  - wasm
+title: rust-libp2p in the Browser with WebRTC!
+description: News about how rust-libp2p in the browser connects to nodes in a network using WebRTC
+date: 2023-12-07
+permalink: ''
+translationKey: ''
+header_image: /rust-libp2p.jpeg
+author: DougAnderson444
+---
+
+# Rust-libp2p WebRTC now available in the Browser
+
+We are excited to announce that [rust-libp2p](https://github.com/libp2p/rust-libp2p) running in the browser can now establish [WebRTC connections](https://webrtc.org/). Before now, WebRTC was only available on the server in [rust-libp2p](https://github.com/libp2p/rust-libp2p), but after months of coding and reviews, we are pleased to annouce it is available on both the server and in the browser from rust-libp2p.
+
+# Why WebRTC for rust-libp2p?
+
+There are a few different Transports to choose from, each with their pros and cons. So, why WebRTC? Let's dive into some of the benefits of WebRTC and why we decided it was important to implement as a Transport within rust-libp2p for use in the browser.
+
+
+## Benefits of WebRTC over WebSockets and WebTransport
+
+There are certain benefits of using WebRTC instead WebSockets and WebTransport.
+
+The benefit of using WebRTC instead of _WebSockets_ on the _server_ is that WebRTC uses self-signed TLS certificates, eliminating the need for users to set up additional services like DNS and Let's Encrypt, which we would have to do with WebSockets to make it available via `wss://`. This is great because it means users can run a public rust-libp2p WebRTC server out of the box and have Transport access to that server from outside their network without additional steps or resources.
```

Comment:
The start of this sentence is a bit repetitive with the use of "benefit".

I'd suggest to highlight both benefits explained in this paragraph with a few keywords and then explain them in more detail, something like (feel free to expand, I just focused on what I think are the essential bits of information):

1. Easier node operation: Modern browsers require the use of TLS in many cases. In the case of WebSockets, the certificates need to be signed by a CA that is in the browsers trust-chain which typically means you'll need to set up DNS and Let's Encrypt (or similar). With WebRTC, we can use self-signed certificates, removing the need for DNS and trusted CAs! WebTransport also allows the use of self-signed certificates but `rust-libp2p` does not (yet) support WebTransport as a server-side transport. Subscribe to https://github.com/libp2p/rust-libp2p/issues/2993 for updates.
2. Improved connectivity: WebRTC allows us to connect two browsers directly. Typically, connecting two nodes requires one of them to "listen" on one or more interfaces and sharing its listen address with the other node. Browsers however don't provide direct access to a machine's network interfaces. With WebRTC, browsers expose an API to establish a direct connection to another browser with the help of a signalling server.

Reply:
I like it! I'll re-word this a bit to focus on these two areas